src/main/java/graphql/execution/ResolveType.java

```diff
@@ -20,33 +20,24 @@
 public class ResolveType {
 
 
-    public GraphQLObjectType resolveType(ExecutionContext executionContext, MergedField field, Object source, ExecutionStepInfo executionStepInfo, GraphQLType fieldType) {
+    public GraphQLObjectType resolveType(ExecutionContext executionContext, MergedField field, Object source, ExecutionStepInfo executionStepInfo, GraphQLType fieldType, Object localContext) {
         GraphQLObjectType resolvedType;
+        DataFetchingFieldSelectionSet fieldSelectionSet = buildSelectionSet(executionContext, field, (GraphQLOutputType) fieldType, executionStepInfo);
+        TypeResolutionEnvironment env = TypeResolutionParameters.newParameters()
+                .field(field)
+                .fieldType(fieldType)
+                .value(source)
+                .argumentValues(executionStepInfo.getArguments())
+                .selectionSet(fieldSelectionSet)
+                .context(executionContext.getContext())
+                .graphQLContext(executionContext.getGraphQLContext())
+                .localContext(localContext)
+                .schema(executionContext.getGraphQLSchema())
+                .build();
         if (fieldType instanceof GraphQLInterfaceType) {
```

Comment:
Simplified - there was never a need for the union vs interface type - there are never treated special outside this spot and again only for resolver lookup